compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/LangLibrary.java

```diff
@@ -0,0 +1,225 @@
+            case INT:
+            case BYTE:
+                return INT.getName();
+            case FLOAT:
+                return FLOAT.getName();
+            case DECIMAL:
+                return DECIMAL.getName();
+            case STRING:
+                return STRING.getName();
+            case BOOLEAN:
+                return BOOLEAN.getName();
+            case ARRAY:
+            case TUPLE:
+                return ARRAY.getName();
+            case STREAM:
+                return STREAM.getName();
+            case OBJECT:
+                return OBJECT.getName();
+            case RECORD:
+            case MAP:
+                return MAP.getName();
+            case ERROR:
+                return ERROR.getName();
+            case FUTURE:
+                return FUTURE.getName();
+            case TYPEDESC:
+                return TYPEDESC.getName();
+            case XML:
+                return XML.getName();
+            case TABLE:
+                return TABLE.getName();
+            default:
+                return "value";
+        }
+    }
+
+    private static Map<String, Map<String, BInvokableSymbol>> getLangLibMethods(Map<String, BPackageSymbol> langLibs) {
+        Map<String, Map<String, BInvokableSymbol>> langLibMethods = new HashMap<>();
+
+        for (Map.Entry<String, BPackageSymbol> entry : langLibs.entrySet()) {
+            String key = entry.getKey();
+            BPackageSymbol value = entry.getValue();
+
+            Map<String, BInvokableSymbol> methods = new HashMap<>();
+
+            for (Map.Entry<Name, Scope.ScopeEntry> nameScopeEntry : value.scope.entries.entrySet()) {
+                BSymbol symbol = nameScopeEntry.getValue().symbol;
+
+                if (symbol.kind != SymbolKind.FUNCTION) {
+                    continue;
+                }
+
+                BInvokableSymbol invSymbol = (BInvokableSymbol) symbol;
+
+                if (Symbols.isFlagOn(invSymbol.flags, Flags.PUBLIC) && !invSymbol.params.isEmpty() &&
+                        key.compareToIgnoreCase(invSymbol.params.get(0).type.getKind().name()) == 0) {
+                    methods.put(invSymbol.name.value, invSymbol);
+                }
+            }
+
+            langLibMethods.put(key, methods);
+        }
+
+        populateLangValueLibrary(langLibs, langLibMethods);
+        return langLibMethods;
+    }
+
+    private static void populateLangValueLibrary(Map<String, BPackageSymbol> langLibs,
+                                                 Map<String, Map<String, BInvokableSymbol>> langLibMethods) {
+        BPackageSymbol langValue = langLibs.get(LANG_VALUE);
+        Map<String, BInvokableSymbol> methods = new HashMap<>();
+
+        for (Map.Entry<Name, Scope.ScopeEntry> nameScopeEntry : langValue.scope.entries.entrySet()) {
+            BSymbol symbol = nameScopeEntry.getValue().symbol;
+
+            if (symbol.kind != SymbolKind.FUNCTION || !Symbols.isFlagOn(symbol.flags, Flags.LANG_LIB)) {
```

Comment:
Can there be symbols in the `lang.value` module without the `LANG_LIB` flag?

Reply:
There can be I guess. Currently we have added the `LANG_LIB` flag to lang lib functions. I guess we can get rid of the flag check

Reply:
Checked this. Forgot that I explicitly checked the lang lib flag to filter out the module init, start, stop functions.